### PR TITLE
MLS: correct typo in `principle` by using `principal` instead

### PR DESCRIPTION
**Pull request by keineahnung2345**

### CODE REVIEW

**Comment by mvieth** [comment]
It is problematic to just change the API. You could add a function with the old name that calls the function with the new, correct name, and add a deprecation notice.

File: surface/include/pcl/surface/mls.h [hunk 1]
```diff
@@ -135,13 +135,23 @@ namespace pcl
...
     inline Eigen::Vector2f
+    calculatePrincipalCurvatures (const double u, const double v) const;
+
+    /** \brief Calculate the principal curvatures using the polynomial surface.
+      * \param[in] u The u-coordinate of the point in local MLS frame.
+      * \param[in] v The v-coordinate of the point in local MLS frame.
+      * \return The principal curvature [k1, k2] at the provided ub coordinates.
+      * \note If an error occurs the MLS_MINIMUM_PRINCIPAL_CURVATURE is returned.
+      */
+    PCL_DEPRECATED(1, 13, "use calculatePrincipalCurvatures() instead")
```
Comment (kunaltyagi) [inline surface/include/pcl/surface/mls.h:153]: Current version + 3 (deprecation policy is removal in 2 releases), that makes it 14. First notice of deprecation in 1.12 and removal in 1.14

**Comment by keineahnung2345** [comment]
Another question, the constant `MLS_MINIMUM_PRINCIPAL_CURVATURE` is mentioned in the comment, but I can't find it in the code, should I create this constant and replace the `1e-5` in `calculatePrincipalCurvatures` with it?

**Comment by mvieth** [comment]
Hm, the constant was added in [this commit](https://github.com/PointCloudLibrary/pcl/pull/1960/commits/bb532955896c706c43d3ea4a18ef5e2dd560e823), but later removed in the same pull request, and only the mention in the documentation remained. I kinda agree with the review comments on the addition, so I would tend to just change the documentation.

**Comment by keineahnung2345** [comment]
@mvieth I've updated the doc.

File: surface/include/pcl/surface/mls.h [hunk 2]
```diff
@@ -135,13 +135,23 @@ namespace pcl
     inline PolynomialPartialDerivative
     getPolynomialPartialDerivative (const double u, const double v) const;
 
-    /** \brief Calculate the principle curvatures using the polynomial surface.
+    /** \brief Calculate the principal curvatures using the polynomial surface.
       * \param[in] u The u-coordinate of the point in local MLS frame.
       * \param[in] v The v-coordinate of the point in local MLS frame.
       * \return The principle curvature [k1, k2] at the provided uv coordinates.
       * \note If an error occurs the MLS_MINIMUM_PRINCIPLE_CURVATURE is returned.
```
Comment (mvieth) [inline surface/include/pcl/surface/mls.h:142]:
```suggestion
      * \note If an error occurs then 1e-5 is returned.
```

Comment (keineahnung2345) [reply surface/include/pcl/surface/mls.h]: Ok, thanks!

**Review by mvieth — APPROVED** [review]

**Review by kunaltyagi — APPROVED** [review]

File: surface/include/pcl/surface/mls.h [hunk 3]
```diff
@@ -135,13 +135,23 @@ namespace pcl
...
     inline Eigen::Vector2f
+    calculatePrincipalCurvatures (const double u, const double v) const;
+
+    /** \brief Calculate the principal curvatures using the polynomial surface.
+      * \param[in] u The u-coordinate of the point in local MLS frame.
+      * \param[in] v The v-coordinate of the point in local MLS frame.
+      * \return The principal curvature [k1, k2] at the provided ub coordinates.
+      * \note If an error occurs then 1e-5 is returned.
+      */
+    PCL_DEPRECATED(1, 14, "use calculatePrincipalCurvatures() instead")
```
Comment (kunaltyagi) [inline surface/include/pcl/surface/mls.h:153]:
```suggestion
    PCL_DEPRECATED(1, 15, "use calculatePrincipalCurvatures() instead")
```

**Review by kunaltyagi — COMMENTED** [review]
Can we move the deprecated function to the header? That would make it easier to remove in future

**Comment by keineahnung2345** [comment]
@kunaltyagi Sorry but what do you mean by "move to the declaration site"?

**Comment by kunaltyagi** [comment]
The simple definition of the deprecated function could be moved to the header (the function declaration) so removing it post deprecation is easier.

**Comment by mvieth** [comment]
Hm, now macOS fails because of the undefined inline function. Is there any other solution than to revert the last commit?

**Comment by keineahnung2345** [comment]
According to https://github.com/mpx/lua-cjson/issues/28, one way is removing `inline`...

**Comment by mvieth** [comment]
You mean removing `inline` from `calculatePrincipalCurvatures`? Sure, let's give it a try.

**Comment by keineahnung2345** [comment]
@mvieth It works.

**Review by mvieth — APPROVED** [review]

**Review by kunaltyagi — APPROVED** [review]

**Pull request merged**